lib/drive_synchronizer.rb

```diff
@@ -3,22 +3,27 @@ class DriveSynchronizer < Synchronizer
 
     protected
     def perform_sync
-      session = GoogleDrive::Session.from_service_account_key(StringIO.new(@api_key))
+      session = session_meth
+      full_path = backup.path
+      filename = backup.filename
+      file = session.upload_from_file(full_path, filename)
+      add_to_folder(session, file)
+      session.root_collection.remove(file)
+    end
+
+    def session_meth
```

Comment:
https://github.com/berlindiamonds/discourse-googledrive-backup/blob/31391e9d32afe23f29121be737b60d1b2d9175d7/lib/drive_synchronizer.rb

just added @session to the constructor